internal/corazawaf/transaction.go

```diff
@@ -1520,37 +1522,38 @@ type TransactionVariables struct {
 	responseXML          collection.Map
 	multipartPartHeaders collection.Map
 	// Persistent variables
-	ip collection.Map
+	ip collection.Noop
 }
 
 func NewTransactionVariables() *TransactionVariables {
 	v := &TransactionVariables{}
+	v.noop = collections.NewNoop()
 	v.urlencodedError = collections.NewSingle(variables.UrlencodedError)
 	v.responseContentType = collections.NewSingle(variables.ResponseContentType)
 	v.uniqueID = collections.NewSingle(variables.UniqueID)
-	v.authType = collections.NewSingle(variables.AuthType)
+	v.authType = collections.NewNoop()
 	v.filesCombinedSize = collections.NewSingle(variables.FilesCombinedSize)
-	v.fullRequest = collections.NewSingle(variables.FullRequest)
+	v.fullRequest = collections.NewNoop()
 	v.fullRequestLength = collections.NewSingle(variables.FullRequestLength)
 	v.inboundDataError = collections.NewSingle(variables.InboundDataError)
 	v.matchedVar = collections.NewSingle(variables.MatchedVar)
 	v.matchedVarName = collections.NewSingle(variables.MatchedVarName)
-	v.multipartBoundaryQuoted = collections.NewSingle(variables.MultipartBoundaryQuoted)
-	v.multipartBoundaryWhitespace = collections.NewSingle(variables.MultipartBoundaryWhitespace)
-	v.multipartCrlfLfLines = collections.NewSingle(variables.MultipartCrlfLfLines)
+	v.multipartBoundaryQuoted = collections.NewNoop()
+	v.multipartBoundaryWhitespace = collections.NewNoop()
+	v.multipartCrlfLfLines = collections.NewNoop()
 	v.multipartDataAfter = collections.NewSingle(variables.MultipartDataAfter)
-	v.multipartDataBefore = collections.NewSingle(variables.MultipartDataBefore)
-	v.multipartFileLimitExceeded = collections.NewSingle(variables.MultipartFileLimitExceeded)
-	v.multipartHeaderFolding = collections.NewSingle(variables.MultipartHeaderFolding)
-	v.multipartInvalidHeaderFolding = collections.NewSingle(variables.MultipartInvalidHeaderFolding)
-	v.multipartInvalidPart = collections.NewSingle(variables.MultipartInvalidPart)
-	v.multipartInvalidQuoting = collections.NewSingle(variables.MultipartInvalidQuoting)
-	v.multipartLfLine = collections.NewSingle(variables.MultipartLfLine)
-	v.multipartMissingSemicolon = collections.NewSingle(variables.MultipartMissingSemicolon)
-	v.multipartStrictError = collections.NewSingle(variables.MultipartStrictError)
-	v.multipartUnmatchedBoundary = collections.NewSingle(variables.MultipartUnmatchedBoundary)
+	v.multipartDataBefore = collections.NewNoop()
```

Comment:
Let's remove these fields completely and have the getter methods return the singleton